File: 알고리즘/정렬/insertion_sort.py
import random

def insertion_sort(data):
    for i in range(1, len(data)):
        for j in range(i, 0, -1):
            if data[j] < data[j - 1]:
                data[j], data[j - 1] = data[j - 1], data[j]
            else:
                break
    return data

data_list: list = random.sample(range(100), 50)

print(insertion_sort(data_list))

File: 알고리즘/정렬/insertion_sort.js
function selectionSort() {
  const array = [80, 30, 90, 40, 50, 70, 60, 20, 10, 100, 50];

  // 가장 최솟값을 찾아야 한다.
  for (let i = 0; i < array.length; i++) {
    const minIdx = i;
    for (let j = i + 1; j < array.length; j++) {
      if (array[j] < array[minIdx]) minIdx = j;
    }

    if (minIdx !== i) {
      [array[i], array[minIdx]] = [array[minIdx], array[i]];
    }
  }
}

console.log(insertionSort());
